python/looker_sdk/rtl/api_settings.py

```diff
@@ -72,6 +73,10 @@ def __init__(
             section (str): section in config file. If not supplied default to
                 reading first section.
         """
+        if not os.path.isfile(filename):
+            if filename and filename not in _DEFAULT_INIS:
+                raise FileNotFoundError(f"No config file found: '{filename}'")
+
```

Comment:
I made config file existence checking eager to simplify the config reading below